### ISSUE TITLE
Ability to manage single entity types as well as entities

### ISSUE THREAD

**Issue by luccamendonca** [hbagdi/deck#150]
Hey Harry!

We talked a bit about migrating from gokong to you `decK/go-kong` previously and after that we started working on using decK as a library on our own CLI.

Now, for our specific use-case we'd like to be able to manipulate individual entities instead of the whole workspace on a state.

For example, we have multiple workspaces with multiple services and our goal is to make our Kong EE a Gateway As A Service for multiple internal customers, where each "customer" may have one or multiple apps inside one or multiple workspaces and that division is not homogeneous.

Now, we tinkered a bit with `decK/go-kong` and had some ideas on how we should be able to manage entities of a single type as well as single entities.

We added `_entity_type` on the yaml file and created a new `export` command, to isolate this "feature".

Of course this was only a *POC* to also help us better understand how `decK/go-kong` works under the hood and we wanted to know what do you think about this approach and, of course, if you have any other possible solutions in mind, like having additional params on the `dump` command instead of creating a new one.

Please, let us know if you have already thought of anything on those lines so we can get our efforts together on improving and adding features to `decK/go-kong` :)

Thanks!

cc @marcelometal

**Comment by hbagdi** [comment]
> Now, we tinkered a bit with decK/go-kong and had some ideas on how we should be able to manage entities of a single type as well as single entities.

Can you elaborate on what you mean by single type and single entities? An example would help as well.

Have you looked at using tags to manage distributed configuration?
It is explained here: https://deck.yolo42.com/guides/distributed-configuration.html

**Comment by luccamendonca** [comment]
Sure thing!

So, let's say a customer X has a couple services they manage under the workspace "testing". Having in mind that there are other services from different customers there as well.

Our main goal is to add the possibility for them to only manage (dump/diff/sync) those specific services, instead of having to dump the whole workspace. The same goes, of course, for consumers, routes and etc

We have created state files containing only one service, for example, but when running the `diff` command we'd see a whole bunch of entities to be **deleted**, because the diff is made by using the whole workspace content instead of only what the file contains.

And, of course, it makes sense considering you chose to delete entities using `sync` instead of having, for example, a dedicated command for deleting them. Given that decK's purpose is to manage/configure Kong as a whole it makes perfect sense.

---

BUT even though we had already made some stuff using the tags filter and it didn't seem, at first, that it would be helpful on our case... we gave it more thought about it and we were probably just addicted to the approach we had before starting to experiment with decK and were trying to do the same thing we had on our internal solution applied to decK.

Anyway, now you know a bit more about why we had this intention at first but it's more likely we'll go with the tags approach.

Thanks for the quick response and sorry to waste your time :)
